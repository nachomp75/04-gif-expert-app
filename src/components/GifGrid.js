import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { PropTypes } from 'prop-types';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p className="animate_animated animate__flash">Loading</p>}
      <div className="card-grid animate__animated animate__fadeIn">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
