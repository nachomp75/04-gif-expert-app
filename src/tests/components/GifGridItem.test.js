import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing GifGridItem', () => {
  const title = 'Un t√≠tulo';
  const url = 'https://localhost/something.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should show the component properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a paragraph with the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should should have an image with the same url and alt from the props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should have animate__zoomIn', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('animate__zoomIn')).toBe(true);
  });
});
