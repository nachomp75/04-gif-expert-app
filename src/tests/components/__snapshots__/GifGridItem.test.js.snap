// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem should show the component properly 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Un título"
    src="https://localhost/something.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
